plutus-ledger-api/src/v1/transaction.rs

```diff
@@ -211,6 +240,21 @@ impl IsPlutusData for TxInInfo {
     }
 }
 
+impl From<(TransactionInput, TransactionOutput)> for TxInInfo {
+    fn from((reference, output): (TransactionInput, TransactionOutput)) -> TxInInfo {
+        TxInInfo { reference, output }
+    }
+}
+
+impl From<(&TransactionInput, &TransactionOutput)> for TxInInfo {
```

Comment:
[nit]: same comment in the v2 module - do we really need this instance? I think we can return something like &'a TxInInfo and make the clone explicit.

Reply:
It was a nice workflow with tx-bakery, when I used `query_utxo`, I got back a `BTree<TransactionInput, TransactionOutput>`, so I could just use `iter().map(TxInInfo::from)` on it. However, I had to make some changes in the query function, so this is not relevant now.